tests/js/client/dump/dump-test.inc

```diff
@@ -1563,6 +1563,38 @@
         assertEqual(graph.orphanCollections.length, 0);
       },
 
+      testSmartGraph3_11_compat: function () {
+      return;
+        if (!isEnterprise) {
+          return;
+        }
+        const smartGraphName = "UnitTestDumpSmartGraphCompat";
+        const graphsColl = getCollection('_graphs');
+        assertTrue(!!graphsColl.exists(smartGraphName));
+
+        const graphDoc = graphsColl.document(smartGraphName);
+        // This is the requirement to test if the compat cid has effect
+        assertTrue(graphDoc.hasOwnProperty('initialCid'));
+        // Now let us try to modify the GraphRelations
+        const newRelationName = `${smartGraphName}NewEdge`;
+        const newVertexName = `${smartGraphName}NewVertex`;
+        const newRelation = gm._relation(newRelationName,
+          "UnitTestDumpSmartVerticesCompat",
+          newVertexName
+        );
+        // New collections do not yet exist, that would be forbidden.
+        assertTrue(db._collections().indexOf(newRelationName) === -1);
+        assertTrue(db._collections().indexOf(newVertexName) === -1);
+        const graph = gm._graph(smartGraphName);
+        graph._extendEdgeDefinitions(newRelation);
+        // New collections are added
+        assertTrue(db._collections().indexOf(newRelationName) !== -1);
+        assertTrue(db._collections().indexOf(newVertexName) !== -1);
```

Comment:
I thought that `db._collections()` returns an array of collection objects, not an array of collection names.
So I think these assertions will always succeed, but they do something different than expected.
I think it should rather be `assertFalse(db._collections().map((c) => c.name()).includes(...));`.
Same for the following assertions.